runtime/src/drivers.rs

```diff
@@ -114,31 +105,110 @@ impl Drivers {
                 .map_err(|_| CaliptraError::RUNTIME_INITIALIZE_DPE_FAILED)?;
             let env = DpeEnv::<CptraDpeTypes> {
                 crypto,
-                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut cert_chain),
+                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut drivers.cert_chain),
             };
-            let dpe = Self::initialize_dpe(env, &mut pcr_bank, locality)?;
-            persistent_data.get_mut().dpe = dpe;
+            let dpe = Self::initialize_dpe(env, &mut drivers.pcr_bank, locality)?;
+            drivers.persistent_data.get_mut().dpe = dpe;
         }
 
+        if reset_reason != ResetReason::ColdReset {
+            let dpe = &drivers.persistent_data.get().dpe;
+            // Find root node by finding the context with parent equal to ROOT_INDEX
+            let root_idx = dpe
+                .contexts
+                .iter()
+                .enumerate()
+                .find(|&(idx, context)| {
+                    context.state != ContextState::Inactive
+                        && context.parent_idx == Context::ROOT_INDEX
+                })
+                .ok_or(CaliptraError::RUNTIME_DPE_VALIDATION_FAILED)?
+                .0;
+            if root_idx >= dpe.contexts.len() {
+                return Err(CaliptraError::RUNTIME_DPE_VALIDATION_FAILED);
+            }
+
+            // Ensure context/TCI tree is well-formed (all nodes reachable from root and no cycles)
+            if !dpe.validate_context_tree(root_idx) {
+                // If SRAM Dpe Instance validation fails, disable attestation
+                let mut result = DisableAttestationCmd::execute(&mut drivers);
+                if result.is_ok() {
+                    cprintln!("Disabled attestation due to DPE validation failure");
+                    drivers.mbox.set_status(MboxStatusE::DataReady);
```

Comment:
I think we want to avoid doing anything with the mailbox when calling disable attestation, since it wasn't invoked from the mailbox. This may require refactoring that command a little.

Reply:
Done